multistep/commonsteps/step_http_server.go

```diff
@@ -111,13 +112,20 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr = s.HTTPAddress
+	httpAddrIface, present := state.GetOk("http_ip")
+	if present && httpAddrIface.(string) != "" {
+		httpAddr = httpAddrIface.(string)
+	}
+
+	ui.Say(fmt.Sprintf("Starting HTTP server on address %s and port %d", httpAddr, s.l.Port))
 
 	// Start the HTTP server and run it in the background
 	server := &http.Server{Addr: "", Handler: s.Handler()}
 	go server.Serve(s.l)
 
-	// Save the address into the state so it can be accessed in the future
+	// Save the address and the port into the state so they can be accessed in the future
+	state.Put("http_ip", httpAddr)
```

Comment:
do you think it could be a good idea to use http_addr here
```suggestion
	state.Put("http_addr", httpAddr)
```
It's probably just sort of a convention but I've always said you listen on an address and you connect to an IP ( maybe there's more scripture on that; trying to look it up )